enhancements/channels-packages-discontinued.md

```diff
@@ -0,0 +1,248 @@
+
+### User Stories
+
+#### Story 1
+
+I am as an Operator Author would like to have an option to clarify to the Operator users that
+a channel or the whole packages will no longer be maintained in future releases so that, users
+are able to plan no longer consume them and be aware of the required changes.
+
+#### Story 2
+
+I am as consumer of the GRCP API would like to have an endpoint to consume properties under
+channels and packages so that, I could implement specific features based on the meta info
+provided by Operator Authors over their packages and channels.
+
+### Implementation Details/Notes/Constraints
+
+The proposed changes are to:
+- extend the GRPC API to support responses with this metadata
+- extend the packagemanifests API to include this metadata
+
+> **Notes:** the exposition of the properties is only supported by FBC.
+> So, it seems like we're going to have to touch the [FBC spec](https://github.com/operator-framework/operator-registry/blob/master/alpha/declcfg/declcfg.go) and
+the [GRPC API](https://github.com/operator-framework/operator-registry/tree/master/pkg/api) itself.
+Although a lot of that code is generated from types that are imported and then scaffolded with the makefile.
+
+Then, once the above items be addressed in the OLM/OPM API the OLM console as any other component 
+and consumer of its API would be able to have access to these meta info to properly deal with 
+the scenarios which motivates this proposal. 
+
+### Risks and Mitigations
+
+The risk here is the same to expose any new endpoints in an API.
+That would mean that if properties would not be allowed in the future for channels and packages
+then these changes would need to be deprecated and planned to be removed.
+
+## Design Details
+
+### Test Plan
+
+The exposition of the properties under channels and packages should be unit tested.
+
+### Graduation Criteria
+
+This feature is in GA upon release.
+
+#### Examples
+
+Following some examples over the proposed usage to address this requirement. However, note that
+by addressing the proposed changes here the mata info under the channels and packages
+could also be used to address further requirements.
+
+- For channels:
+
+```yaml
+properties:
+- type: 'olm.deprecated'
+  value:
+    message: 'This channel is deprecated and will no longer be provided in future releases. Please, update your subscription to get updates from the channel stable.'
+    moreInfoURL: 'https://github.com/myorg/mmy-operator-repo/discussions'
+    replacement:
+      name: 'stable'
+```
+
+- For packages:
+
+```yaml
+properties:
+- type: 'olm.deprecated'
+  value:
+    message: 'This package is deprecated and will no longer be provided in future releases. Please, ensure that you use its Backup Service in favour to move forward and began to use the project SQLite Operator.'
+    moreInfoURL: 'https://github.com/myorg/my-operator-repo/deprecation.md'
+    replacement:
+      name: 'SQLite Operator'
```

Comment:
Should the `name` be the actual package name (e.g. "sqllite-operator") instead of it's description?  It could be leveraged by a UI to enable easier transition and discovery of the replacement package.

I wonder if the `name" should be a programmatic, consistent cross-referenceable artifact (channel name, or package name, depending on context) and then if someone wants some freeform text, use a  `message` in the `replacment` object.

Reply:
It is specified under the proposal section: https://github.com/operator-framework/enhancements/pull/113/files#diff-69c814cbb4e88b1c7d6d45f3d1e5c75c9c7c6b24a34067c16798966ed9c5dcb7R128-R137

The name is the package OR the channel name